GameOfLife/GameOfLife/Iteration.cs

```diff
@@ -0,0 +1,162 @@
+        public void CheckCells(string[,] field)
+        {
+            CheckAlive(field);
+            CheckDead(field);
+        }
+
+        /// <summary>
+        /// Determines if an alive cell dies before the next generation.
+        /// </summary>
+        /// <param name="field"></param>
+        public void CheckAlive(string[,] field)
+        {
+            int neigboursCountOfAlive = 0;
+            bool wrappedX = false;
+            bool wrappedY = false;
+
+            for (int i = 0; i < field.GetLength(0); i++)
+            {
+                for (int j = 0; j < field.GetLength(1); j++)
+                {
+                    if (field[i, j] == "X")
+                    {
+                        for (int neighbourX = i - 1; neighbourX <= i + 1; neighbourX++)
+                        {
+                            for (int neighbourY = j - 1; neighbourY <= j + 1; neighbourY++)
+                            {
+                                if (neighbourX == -1)
+                                {
+                                    neighbourX = field.GetLength(0) - 1;
+                                    wrappedX = true;
+                                }
+                                if (neighbourY == -1)
+                                {
+                                    neighbourY = field.GetLength(1) - 1;
+                                    wrappedY = true;
+                                }
+                                if (field[neighbourX % field.GetLength(0), neighbourY % field.GetLength(1)] == "X")
+                                {
+                                    neigboursCountOfAlive++;
+                                }
+                                if (neighbourX == i && neighbourY == j && neigboursCountOfAlive > 0)
+                                {
+                                    neigboursCountOfAlive--;
+                                }
+                                if (wrappedY)
+                                {
+                                    neighbourY = j - 1;
+                                    wrappedY = false;
+                                }
+                            }
+                            if (wrappedX)
+                            {
+                                neighbourX = i - 1;
+                                wrappedX = false;
+                            }
+                        }
+                        if (neigboursCountOfAlive < 2 || neigboursCountOfAlive > 3)
+                        {
+                            cellsToDieX.Add(i);
+                            cellsToDieY.Add(j);
+                        }
+                        neigboursCountOfAlive = 0;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Determines if a dead cell is reborn before the next generation.
+        /// </summary>
+        /// <param name="field"></param>
+        public void CheckDead(string[,] field)
```

Comment:
Check accessibility modifiers, if method or property is used only within this file it should be private.

Reply:
Changed some "public" access modifiers to "private".

Reply:
You can mark this comment as resolved.
For some reason can't do it myself.